scipy/special/setup.py

```diff
@@ -87,17 +94,21 @@ def configuration(parent_package='',top_path=None):
                          **cfg)
 
     # Extension _ufuncs_cxx
-    ufuncs_cxx_src = ['_ufuncs_cxx.cxx', 'sf_error.c',
+    ufuncs_cxx_src = ['_ufuncs_cxx.cxx',
+                      #'sf_error.c',
```

Comment:
`sf_error.c` I think should be linked in here, and I guess the builds will work after uncommenting. (There are two `.so` files; Python's build system is too bare-bones to know how to link Fortran and c++ into the same library.)

Reply:
I'm wrangling with failure-to-build on OSX with conda and Clang c++ compiler. The c++ source fails to build with `sf_error.c` included because for some strange reason a very ancient compatibility mode is selected, and the compiler doesn't accept c++11 code (such as `constexpr` variables). This problem goes away if the C source is removed from the list of source files here, but then it creates a problem undefined name in the .so file.

This may as well be a problem with build-system setup here. I'm still investigating.

I agree that it's a pain to work with Python's extension build mechanism when mixed-language sources are all over the place!

Reply:
See commit 5ee49a3 -- still trying to make it compatible with the build system on Travis CI, which I suppose should be representative of many usage cases.